include/xtensor/xassign.hpp

```diff
@@ -33,7 +47,12 @@ namespace xt
     void assign_data(xexpression<E1>& e1, const xexpression<E2>& e2, bool trivial);
 
     template <class E1, class E2>
-    void assign_xexpression(xexpression<E1>& e1, const xexpression<E2>& e2);
+    std::enable_if_t<!has_assign_to<E1, E2>::value, void>
+    assign_xexpression(xexpression<E1>& e1, const xexpression<E2>& e2);
+
+    template <class E1, class E2>
+    std::enable_if_t<has_assign_to<E1, E2>::value, void>
+    assign_xexpression(xexpression<E1>& e1, const xexpression<E2>& e2);
```

Comment:
Same remark here, I find the implementation with the `static_if` more concise and the API clearer.

Reply:
if i try with static_if I get compilation errors because `assign_to` doesn't exist on most xexpressions. 
I thought that branch wouldn't get compiled (is that assumption correct?)